sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -414,13 +459,33 @@ Ext.define('SlateTasksTeacher.controller.Dashboard', {
         if (studentTask) {
             me.getTaskEditorForm().updateRecord(studentTask);
             studentTask.set('SkillIDs', me.getSkillsField().getSkills(false, true)); // returnRecords, idsOnly
-            return me.doSaveStudentTask(studentTask, function() {
-                me.getStudentTasksStore().load({
-                    id: studentTask.getId(),
-                    addRecords: true
-                });
-                taskEditor.close();
-                Ext.toast('Student task successfully updated.');
+            return me.doSaveStudentTask(studentTask, function(rec, success, request) {
+                var response = JSON.parse(request.getResponse().responseText),
```

Comment:
@nbey I've commented on the same code 3 times now, perhaps `me.doSaveStudentTask` should be handling reporting failures? Why else create a wrapper for that call within the controller if it's only going to turn two callbacks into one (which it doesn't need to do because `.save()` already has a `callback` option that gets called with a `success` boolean in either case)